source/grid/grid_in.cc

```diff
@@ -1451,14 +1452,72 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
           in >> line;
         }
 
-      // if the next block is of kind $Entities, ignore it
+      // if the next block is of kind $Entities, parse it
       if (line == "$Entities")
         {
-          do
+          unsigned long n_points, n_curves, n_surfaces, n_volumes, n_physicals;
+          int           tag, physical_tag;
+          double        box_min_x, box_min_y, box_min_z, box_max_x, box_max_y,
+            box_max_z;
+
+          // we only care for physical_tag to fill tag_maps
+          in >> n_points >> n_curves >> n_surfaces >> n_volumes;
+          for (unsigned int i = 0; i < n_points; ++i)
             {
-              in >> line;
+              // parse point ids
+              in >> tag >> box_min_x >> box_min_y >> box_min_z >> box_max_x >>
+                box_max_y >> box_max_z >> n_physicals;
+              AssertThrow(n_physicals < 2,
+                          ExcMessage("More than one tag is not supported!"));
+              for (unsigned int j = 0; j < n_physicals; ++j)
+                in >> physical_tag;
+              tag_maps[0][tag] = (n_physicals == 0) ? 0 : physical_tag;
+            }
+          for (unsigned int i = 0; i < n_curves; ++i)
```

Comment:
maybe say something along the lines of "We really don't care about the curves, but we have to parse them because their format is unstructured".

same for the other sections below.

Reply:
Ah, forget it -- you actually do care.